### PR TITLE
intltool: use perl detected by environment

### PR DESCRIPTION
**Pull request by maxim-belkin**
- [x] Have you followed the [guidelines for contributing](https://github.com/Linuxbrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Linuxbrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by maxim-belkin** [comment]
It's trickier than I thought. When `intltool` is installed, it hard-codes `perl` binary that is used during the installation. So, if a user has Homebrew `perl` installed, `intltool` will hard-code Homebrew's `perl`, if Homebrew's perl is not installed then it will hard-code system's perl into itself...

**Comment by maxim-belkin** [comment]
intltool requires
  * perl 5.8.1+
  * XML::Parser

XML::Parser requires
  * expat

**Comment by maxim-belkin** [comment]
should we add
```ruby
depends_on "expat" => :run
```
here?

**Comment by maxim-belkin** [comment]
@rwhogg what do you think?

File: Formula/intltool.rb [hunk 1]
```diff
@@ -18,6 +18,9 @@ def install
     system "./configure", "--prefix=#{prefix}",
                           "--disable-silent-rules"
     system "make", "install"
+    Dir[bin/"intltool-*"].each do |f|
```
Comment (rwhogg) [inline Formula/intltool.rb]: IMO this should be wrapped in an `unless OS.mac?`.

**Comment by rwhogg** [comment]
> should we add
depends_on "expat" => :run
here?

I invite @sjackman and @DoomHammer to disagree with me here, but IMO until we get the perl-modules formula merged we should just have Perl depend on expat (as it does since #1209) and not include it in every formula that depends on XML::Parser.

The downside of that plan is that it introduces weirdness for a system which:
a) uses a distro-provided Perl AND
b) does not have expat installed either through the system or through brew
when installing a formula which depends on intltool but not expat. I'd argue that such systems are both rare and oddly configured, so that's not a huge deal IMO.

Either way, the `:run` is unnecessary IMO, even though we don't need it at build time. My understanding was that `:run` is only for "weird" dependencies like Libtool that are build-time-only in 99% of cases.

**Comment by DoomHammer** [comment]
Agree with @rwhogg about temporary solution. This module is widely used so it might as well go to the main formula.

**Comment by maxim-belkin** [comment]
The problem that I'm trying to fix here is that `intltool` hard-codes paths to perl that is used to install it. The solution is to use `#!/usr/bin/env perl`. Caveat is that  `env` on old systems might accept only one argument (that is `perl`). The problem is that `intltool-*` tools use `/path/to/perl -w`.  `-w` enables warnings and is, actually, deprecated in favor of `use warnings;`. This PR does exactly this: changes shebang to `#!/usr/bin/env perl` and adds `use warnings;`

**Comment by DoomHammer** [comment]
Nice hack! :+1:

**Comment by sjackman** [comment]
I'm happy with this fix. Are you planning to update the Linuxbrew bottle, or ignore it?

**Comment by maxim-belkin** [comment]
I'll update the bottle. I don't know when the next release is coming.

**Comment by sjackman** [comment]
Updating the bottle will requires the Homebrew/Linuxbrew revisions to diverge, which I'd prefer to avoid when possible. Under what circumstances are users affected by this issue?

**Pull request merged**